packages/@lwc/engine/src/framework/stylesheet.ts

```diff
@@ -83,18 +80,40 @@ export function resetStyleAttributes(vm: VM): void {
 
     // Reset the scoping attributes associated to the context.
     context.hostAttribute = context.shadowAttribute = undefined;
+
+    // removing style tag if present
+    if (!isNull(context.styleVNode) && !isUndefined(context.styleVNode)) {
```

Comment:
destruct context to get the style node, btw, this name is incorrect, misleading, this is not a VNode, it is a node, or a styleElement to be more specific.